consent/consent-page/cdr_client.go

```diff
@@ -1,86 +1,150 @@
-							Currency: "USD",
-						},
-					},
-				},
-				{
-					ID:   "1000001",
-					Name: "Savings",
-					Balance: Balance{
-						AccountID: "1000001",
-						Amount: BalanceAmount{
-							Amount:   "150",
-							Currency: "USD",
-						},
-					},
-				},
-				{
-					ID:   "1000002",
-					Name: "Savings 2",
-					Balance: Balance{
-						AccountID: "1000002",
-						Amount: BalanceAmount{
-							Amount:   "175",
-							Currency: "USD",
-						},
-					},
-				},
+func NewCDRBankClient(config Config) *CDRBankClient {
+	var (
+		pool  *x509.CertPool
+		cert  tls.Certificate
+		certs = []tls.Certificate{}
+		data  []byte
+		err   error
+	)
+
+	if config.BankClientConfig.CertFile == "" && config.BankClientConfig.KeyFile == "" {
+		return &CDRBankClient{
+			httpClient: http.DefaultClient,
+			cc: clientcredentials.Config{
+				ClientID:     config.BankClientConfig.ClientID,
+				ClientSecret: config.BankClientConfig.ClientSecret,
+				TokenURL:     config.BankClientConfig.TokenURL,
+				Scopes:       config.BankClientConfig.Scopes,
 			},
-		}, nil
+			bankURL: config.BankClientConfig.URL.String(),
+		}
 	}
 
-	return InternalAccounts{
-		Accounts: []InternalAccount{
-			{
-				ID:   "96565987",
-				Name: "Credit",
-				Balance: Balance{
-					AccountID: "96565987",
-					Amount: BalanceAmount{
-						Amount:   "100",
-						Currency: "USD",
-					},
-				},
-			},
-			{
-				ID:   "1122334455",
-				Name: "Savings",
-				Balance: Balance{
-					AccountID: "1122334455",
-					Amount: BalanceAmount{
-						Amount:   "150",
-						Currency: "USD",
-					},
+	if cert, err = tls.LoadX509KeyPair(config.BankClientConfig.CertFile, config.BankClientConfig.KeyFile); err != nil {
+		logrus.Fatalf("failed to read certificate and private key %v", err)
+	}
+	certs = append(certs, cert)
+
+	if pool, err = x509.SystemCertPool(); err != nil {
+		logrus.Fatalf("failed to read system root CAs %v", err)
+	}
+
+	if data, err = os.ReadFile(config.RootCA); err != nil {
+		logrus.Fatalf("failed to read http client root ca: %v", err)
+	}
+	pool.AppendCertsFromPEM(data)
+
+	return &CDRBankClient{
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+				TLSClientConfig: &tls.Config{
+					RootCAs:      pool,
+					MinVersion:   tls.VersionTLS12,
+					Certificates: certs,
 				},
 			},
 		},
-	}, nil
+		cc: clientcredentials.Config{
+			ClientID:     config.BankClientConfig.ClientID,
+			ClientSecret: config.BankClientConfig.ClientSecret,
+			TokenURL:     config.BankClientConfig.TokenURL,
+			Scopes:       config.BankClientConfig.Scopes,
+		},
+		bankURL: config.BankClientConfig.URL.String(),
+	}
+}
+
+func (c *CDRBankClient) GetInternalAccounts(id string) (InternalAccounts, error) {
+	var (
+		token    *oauth2.Token
+		request  *http.Request
+		response *http.Response
+		err      error
+	)
+
+	if token, err = c.cc.Token(context.WithValue(context.Background(), oauth2.HTTPClient, c.httpClient)); err != nil {
```

Comment:
You should pass ctx to this function.
Additionally, if this client is only used to call APIs that are client-credentials protected, it's possible to preconfigure HTTP client to automatically issue / manage access tokens before calling APIs.